### DISCUSSION TITLE
MagickImage cannot open NEF files captured by Nikon Z9

### DISCUSSION THREAD

**Discussion by lt090004114** [ImageMagick/ImageMagick#7044] [Help]
### ImageMagick version

13.5.0

### Operating system, version and so on

windows 10

### Description

 Recently, I converted NEF files into JPG and displayed them on a webpage through bulk code conversion. Discovered that the NEF file captured by Nikon Z9 cannot be read.

**Comment by snibgo** [comment]
IM usually delegates reading NEF files to libraw. On their page, https://www.libraw.org/download , they say "Nikon Z9: standard compression formats only". On https://www.libraw.org/supported-cameras, they say Z9 "HE/HE* formats are not supported yet".

EDIT: IM can certainly read some raw Z9 files, such as https://www.dpreview.com/sample-galleries/4324513839/nikon-z9-firmware-2-0-sample-gallery/5444562552

**Reply by lt090004114** [reply]
Emmm. actually,
using (MagickImage image = new MagickImage(nefPath))
When I use MagickImage to read NEF, error: Unsupported file format or not RAW file `xxx.NEF' @ error/dng.c/ReadDNGImage/574
I found that some NEF files ok, some error.

**Comment by snibgo** [comment]
I tried it with IM v7.1.1-20, which includes libraw v0.21.1. If your versions are older, then I suggest you upgrade.

What does exiftool say about a file you can't read? If it is HE or HE*, then that's the problem.

**Reply by lt090004114** [reply]
I download the neweast version. But change error.
![Uploading im_error.png…]()

**Reply by lt090004114** [reply]
![im_error](https://github.com/ImageMagick/ImageMagick/assets/27264096/ae4f35e5-be97-4009-bb0c-21d5fb5c00e3)
